From c2eb321387c0d01abb4f3434a74178de3a219f7d Mon Sep 17 00:00:00 2001
From: Jiaxing Hu <huhuvmb88@outlook.com>
Date: Sat, 6 Jun 2026 13:56:33 +1200
Subject: [PATCH 5/6] plat-rockchip: rk3576: force CFG_CRYPTO_WITH_CE=y

Cortex-A55 and Cortex-A72 both implement ARMv8 Cryptographic
Extensions (AES, SHA-1, SHA-256, SHA-512).  Force CE-accelerated
crypto as RK3588 does.

Tested on Radxa Rock 4D: check_crypto_extensions() passes, xtest
113/1 (baseline unchanged).

Signed-off-by: Jiaxing Hu <huhuvmb88@gmail.com>
---
 core/arch/arm/plat-rockchip/conf.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/plat-rockchip/conf.mk b/core/arch/arm/plat-rockchip/conf.mk
index e1fb66f00..739f21194 100644
--- a/core/arch/arm/plat-rockchip/conf.mk
+++ b/core/arch/arm/plat-rockchip/conf.mk
@@ -87,6 +87,7 @@ ifeq ($(PLATFORM_FLAVOR),rk3576)
 include core/arch/arm/cpu/cortex-armv8-0.mk
 $(call force,CFG_TEE_CORE_NB_CORE,8)
 $(call force,CFG_AUTO_MAX_PA_BITS,y)
+$(call force,CFG_CRYPTO_WITH_CE,y)
 
 CFG_TZDRAM_START ?= 0x70000000
 CFG_TZDRAM_SIZE  ?= 0x02000000
-- 
2.43.0

